experimental/entity/entity.go

```diff
@@ -0,0 +1,56 @@
+package entity
+
+// This is the base class for (non-raw) kinds
+type Envelope struct {
+	// UID may contain / to indicate parent folder's
+	UID string `json:"UID,omitempty"`
+
+	// dash, ds, alert, folder, svg, png, df, dqr, ... (will validate body)
+	Kind string `json:"kind,omitempty"`
+
+	// Version of the included body (semver)
+	SchemaVersion string `json:"schemaVersion,omitempty"`
+
+	// Entity name
+	Name string `json:"name,omitempty"`
+
+	// Entity description
+	Description string `json:"description,omitempty"`
+
+	// Tags to add for search
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Optional metadata describing where the body came from
+	Provenance *Provenance `json:"provenance,omitempty"`
+}
+
+// Extension to the core entity wrapper that supports managed secure keys
+type EnvelopeWithSecureKeys struct {
+	Envelope
+
+	// NOTE: Although APIs will limit exposing this to people without permissions,
+	// The contents of this map (key+value) may be stored in externally secured object stores (S3, disk, etc)
+	// The values should be lookup keys into a secret service
+	SecureKeys map[string]string `json:"secureKeys,omitempty"`
```

Comment:
🤔 ha -- i think it is called "keys" because it was historically used to save... wait for it... API keys!

But yes, we should call it "SecureValues"!